sdk/src/main/java/com/google/cloud/dataflow/sdk/io/OffsetBasedSource.java

```diff
@@ -308,6 +314,19 @@ public Double getFractionConsumed() {
     }
 
     @Override
+    public long getParallelismConsumed() {
+      return rangeTracker.getProcessedSplitPoints();
+    }
+
+    @Override
+    public long getParallelismRemaining() {
+      if (getCurrentOffset() + 1 >= rangeTracker.getStopPosition()) {
```

Comment:
The current position can be >= the stop position when we're emitting non-split-point records. I assume it's useful to know that we're on the last split.

As for why +1: this is to determine if we're at the last record. If we're on record 99 and the stop offset is 100, we did not meet the conditions you described. The only way I know of to distinguish 99 from 98 (or 1) is to + 1 for the check.